packages/subgraph/src/QFIMapping.ts

```diff
@@ -0,0 +1,572 @@
+import { BigInt, log, store } from "@graphprotocol/graph-ts"
+import { OwnershipTransferred } from "../generated/GrantRoundFactory/GrantRoundFactory"
+import {
+    QFI as QFIContract,
+    ContributionWithdrawn,
+    DeployGrantRound,
+    GrantRoundFinalized,
+    Init,
+    InitQfi,
+    MergeStateAq,
+    SignUp,
+    Contribution as ContributionEvent,
+    VotingPeriodClosed,
+    PreRoundContributionPeriodStarted,
+    PollProcessorAndTallyerChanged,
+    MaciDeployed,
+    QfiDeployed,
+    FundingSourceAdded,
+    FundingSourceRemoved
+} from "../generated/QFI/QFI"
+import { GrantRound as GrantRoundContract } from "../generated/templates/GrantRound/GrantRound"
+import {
+    QFI as QFISchema,
+    GrantRound,
+    PublicKey,
+    Coordinator,
+    Contribution,
+    Contributor,
+    FundingSource,
+    GrantRoundContributor
+} from "../generated/schema"
+import { GrantRound as GrantRoundTemplate } from '../generated/templates'
+
+/**
+ * Handle a smart contract based on MACI (constructor event).
+ * @param event Ethereum event emitted when deploying a smart contract based on MACI.
+ */
+export function handleMaciDeployed(event: MaciDeployed): void {
+    log.debug(`MACI Deployed event block: {}`, [event.block.number.toString()])
+
+    const timestamp = event.block.timestamp.toString()
+    // Get the QFI/MACI instance.
+    const qfiAddress = event.address
+    const qfiContract = QFIContract.bind(qfiAddress)
+    const qfiId = qfiAddress.toHexString()
+    const qfi = new QFISchema(qfiId)
+
+    qfi.stateAqAddress = event.params._stateAq
+    qfi.pollFactoryAddress = event.params._pollFactory
+    qfi.initialVoiceCreditProxyAddress = event.params._initialVoiceCreditProxy
+    qfi.signUpGatekeeperAddress = event.params._signUpGatekeeper
+    qfi.signUpTimestamp = event.params._timestamp
+    qfi.numSignUps = new BigInt(0)
+    qfi.nextGrantRoundId = new BigInt(0)
+    qfi.createdAt = timestamp
+    qfi.lastUpdatedAt = timestamp
+
+    // External call to contract instance (one time event - good trade-off for sync time).
+    qfi.stateTreeDepth = qfiContract.stateTreeDepth()
+    qfi.numSignUps = qfiContract.numSignUps()
+
+    qfi.save()
+}
+
+/**
+ * Handle a new QFI instance deploy.
+ * @param event Ethereum event emitted when a new QFI instance is deployed.
+ */
+export function handleQfiDeployed(event: QfiDeployed): void {
+    log.debug(`QFI Deployed event block: {}`, [event.block.number.toString()])
+
+    // Get the QFI/MACI instance.
+    const qfiAddress = event.address
+    const qfiId = qfiAddress.toHexString()
+    const qfi = new QFISchema(qfiId)
```

Comment:
I am currently treating QFI as a Data Source, not as a Template in the `subgraph.yaml`. 
Should I manage QFI as a Template the same way I do for GrantRounds? (Also, should I do the same thing with OptimisticRecipientRegistry and GrantRoundFactory? Because they are also treated as Data Sources too) Thanks!